### PR TITLE
Quick bug fix in random_walk.py

### PR DESCRIPTION
**Pull request by b1quint**
The ``random_walk_azel_by_time`` function returns four values, but the ``for`` loop expected only three values.

### CODE REVIEW

File: tests/maintel/tma/test_random_walk.py [hunk 1]
```diff
@@ -121,13 +121,13 @@ def actualPosition(self):
...
-            async for i, az, el, off in self.script.random_walk_azel_by_time():
-                sky_offsets.append(off)
+            async for data in self.script.random_walk_azel_by_time():
+                sky_offsets.append(data.offset)
 
                 # TODO: This should not be necessary
                 # However, if I remove it the loop keeps running forever
                 # It does not seem to happen when running the loop w/ hardware
-                if i == 50:
+                if data.counter == 50:
```
Comment (r-owen) [inline tests/maintel/tma/test_random_walk.py:130]: Regarding your TODO: I would fully expect this to run forever unless `random_walk_azel_by_time` has an intrinsic limit in the # of targets it will return. 

If it does, then this tests shows that limit is broken. if it does not, then you need to decide how many targets you want.

Please remove the TODO comment -- after fixing the method, if necessary.

File: python/lsst/ts/externalscripts/maintel/tma/random_walk.py [hunk 2]
```diff
@@ -201,15 +232,17 @@ def set_metadata(self, metadata):
         metadata.duration = self.config.total_time
 
     async def run(self):
-        async for i, az, el in self.random_walk_azel_by_time():
-            await self.checkpoint(f"[{i}] Tracking {az=}/{el=}.")
-            await self.slew_and_track(az, el, target_name=f"random_walk_{i}")
+        async for data in self.random_walk_azel_by_time():
```
Comment (r-owen) [inline python/lsst/ts/externalscripts/maintel/tma/random_walk.py:235]: What does the "by_time" in the method name mean? This seems to pick a random az/el, but I don't see use of time in that method. Perhaps consider a name such as `get_random_walk_azel_target`?

File: tests/maintel/tma/test_random_walk.py [hunk 3]
```diff
@@ -121,13 +121,13 @@ def actualPosition(self):
             )
 
             sky_offsets = []
-            async for i, az, el, off in self.script.random_walk_azel_by_time():
-                sky_offsets.append(off)
+            async for data in self.script.random_walk_azel_by_time():
```
Comment (r-owen) [inline tests/maintel/tma/test_random_walk.py:124]: Way up above in test_run, about line 70 I see you use a mock for random_walk_azel_by_time. That explains why test_run passed even when you broke the script. You also mock slew_and_track, which means this test is testing essentially none of your code.

I suggest mocking the "tcs" instead, if practical, though that will clearly be more work. (I also asked Tiago to consider adding a simulation mode to these classes and copied you on the ticket.)

**Review by r-owen — COMMENTED** [review]
Looks very good. I am hoping you can beef up the unit test of the run function before final review; let me know what you think.

File: python/lsst/ts/externalscripts/maintel/tma/random_walk.py [hunk 4]
```diff
@@ -262,14 +295,17 @@ async def random_walk_azel_by_time(self):
 
             t = Time.now().to_value("isot")
             self.log.info(
-                f"{t:25s}{step:10d}{current_az:10.2f}{new_az:10.2f}"
+                f"{t:25s}{counter:10d}{current_az:10.2f}{new_az:10.2f}"
```
[the remark this answers is not in the document]
Comment (b1quint) [reply python/lsst/ts/externalscripts/maintel/tma/random_walk.py:298]: This format is useful when running on a Jupyter Notebook because the log messages do not display timestamps. It also displayed the data as a table. However, I agree that we are in a different context here and should use an appropriate format. I applied the suggested changes.

[the remark this answers is not in the document]
Comment (r-owen) [reply python/lsst/ts/externalscripts/maintel/tma/random_walk.py:298]: I very strongly recommend fixing the notebooks to display the date in log messages. If configuring the logger to print to stdout you can specify the format. If printing SAL log messages you can print data.private_sndStamp (after converting to ISO format, if you like -- I usually don't bother because I care more about time differences, which I find easier to compute from floats).

File: python/lsst/ts/externalscripts/maintel/tma/random_walk.py (hunk 2 above)
Comment (b1quint) [reply python/lsst/ts/externalscripts/maintel/tma/random_walk.py:233]: The "by_time" means this function will generate a random target for a finite time. I will update the docstring to try to be more explicit. Initially, the time was an input parameter, making it easier to see, but Tiago asked to use the configuration directly inside the method.

File: tests/maintel/tma/test_random_walk.py (hunk 3 above)
Comment (b1quint) [reply tests/maintel/tma/test_random_walk.py:120]: I **think** I covered what you suggested.

File: tests/maintel/tma/test_random_walk.py (hunk 1 above)
Comment (b1quint) [reply tests/maintel/tma/test_random_walk.py]: The unit test is running fine now.

**Comment by tribeiro** [comment]
Please, make sure you update the version history. These changes should be part of v0.25.0. I will hold on release until we get this PR and #96 merged.

File: doc/version_history.rst [hunk 5]
```diff
@@ -6,6 +6,17 @@
 Version History
 ===============
 
+v0.25.1
```
Comment (tribeiro) [inline doc/version_history.rst:9]: This should be `v0.25.0`

**Review by tribeiro — APPROVED** [review]
Looks good.. Please, fix the CI errors before merging.. It is currently failing isort in your Script.

**Review by tribeiro — CHANGES_REQUESTED** [review]
Please, make these changes part of v0.25.0

**Review by tribeiro — APPROVED** [review]
Looks good! thanks
